galasa-ui/src/components/test-runs/TableDesignRow.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+"use client";
+import styles from "@/styles/TestRunsPage.module.css";
+import { Checkbox } from "@carbon/react";
+import { useSortable } from "@dnd-kit/sortable";
+import { CSS } from "@dnd-kit/utilities";
+import { IconButton } from "@carbon/react";
+import { ChevronDownOutline, ChevronUpOutline, Draggable } from "@carbon/icons-react";
+import { RESULTS_TABLE_COLUMNS } from "@/utils/constants/common";
+import { useTranslations } from "next-intl";
+
+interface TableDesignRowProps {
+  rowId: string;
+  index: number;
+  isSelected: boolean;
+  onSelect: (rowId: string) => void;
+  onClickArrowUp: () => void;
+  onClickArrowDown: () => void;
+}
+
+export default function TableDesignRow({ rowId, index, isSelected,  onSelect, onClickArrowUp, onClickArrowDown }: TableDesignRowProps) {
+  const translations = useTranslations("TableDesignRow");
+
+  const {
+    attributes,
+    listeners,
+    setNodeRef,
+    transform,
+    transition,
+    isDragging,
+  } = useSortable({ id: rowId }); 
+
+  const style = {
+    transform: CSS.Transform.toString(transform),
+    transition,
+    opacity: isDragging ? 0.5 : 1, 
+  };
```

Comment:
Instead of inline styling, we could use CSS classes for the different states and apply them conditionally?